File: codesignal/arcade/Intro/JS/sortByHeight.js
function sortByHeight(a) {
    let values = []
    
    for(let i = 0; i < a.length; i++){
        if(a[i] != -1) values.push(a[i])
    }
    
    values.sort((b, c) => b - c)
    
    for(let i = 0; i < a.length; i++){
        if(a[i] != -1) a[i] = values.shift()
    }
    
    return a
}

File: codesignal/arcade/Intro/Csharp/sortByHeight.cs

int[] sortByHeight(int[] a) {
    
    List<int> values = new List<int>();
    
    for(int i = 0; i < a.Length; i++){
        if(a[i] != -1) values.Add(a[i]);
    }
    
    values.Sort();
    
    int pos = 0;
    
    for(int i = 0; i < a.Length; i++){
        if(a[i] != -1){
            a[i] = values[pos];
            pos++;
        }
    }
    
    return a;
}